apps/condo/domains/miniapp/access/B2BAppContext.js

```diff
@@ -9,44 +9,54 @@ const { getById } = require('@open-condo/keystone/schema')
 
 const { checkB2BAppAccessRight } = require('@condo/domains/miniapp/utils/accessSchema')
 const { checkOrganizationPermission } = require('@condo/domains/organization/utils/accessSchema')
-const { SERVICE } = require('@condo/domains/user/constants/common')
+const { SERVICE, STAFF } = require('@condo/domains/user/constants/common')
+const { canDirectlyReadSchemaObjects, canDirectlyManageSchemaObjects } = require('@condo/domains/user/utils/directAccess')
 
 
 /**
  * B2B App context may only be read:
  * 1. Admin / support
- * 2. Organization employee (without settings / state)
- * 3. Integration service account
+ * 2. Users with direct access
+ * 3. Organization employee (without settings / state)
+ * 4. Integration service account
  */
-async function canReadB2BAppContexts ({ authentication: { item: user } }) {
+async function canReadB2BAppContexts ({ authentication: { item: user }, listKey }) {
     if (!user) return throwAuthenticationError()
     if (user.deletedAt) return false
 
     if (user.isAdmin || user.isSupport) return {}
 
+    const hasDirectAccess = await canDirectlyReadSchemaObjects(user, listKey)
+    if (hasDirectAccess) return true
```

Comment:
May be we can return not only true/false but also can use it for read restrictions?

Reply:
Is there a real case when you need filtering? 

Since right now (webhooks / dev-portal user / support) will need access to entire table.